app/quest-boost/[boostId]/page.tsx

```diff
@@ -155,44 +115,30 @@ export default function Page({ params }: BoostQuestPageProps) {
           <p>Reward:</p>
           <div className="flex flex-row gap-2">
             <p className={styles.claim_button_text_highlight}>
-              {boost?.amount} USDC
+              {boost?.amount} {getTokenName(boost?.token ?? "")}
             </p>
-            <CDNImage
-              src={"/icons/usdc.svg"}
-              priority
-              width={32}
-              height={32}
-              alt="usdc icon"
-            />
+            <TokenSymbol tokenAddress={boost?.token ?? ""} />
           </div>
           <p>among</p>
           <p className={styles.claim_button_text_highlight}>
             {participants} players
           </p>
         </div>
-        <div>
-          <Button
-            disabled={
-              boost?.claimed ||
-              hexToDecimal(boost?.winner ?? "") !== hexToDecimal(address)
-            }
-            onClick={handleClaimClick}
-          >
-            {(() => {
-              if (boost?.claimed) {
-                return "Claimed ✅";
-              } else if (
-                hexToDecimal(boost?.winner ?? "") === hexToDecimal(address)
-              ) {
-                return "Claim boost reward 🎉 ";
-              } else if (boost && boost?.expiry > Date.now()) {
-                return "Boost has not ended ⌛";
-              } else {
-                return "You’re not selected 🙁";
+        {address ? (
+          <div>
+            <Button
+              disabled={
+                boost &&
+                (boost?.expiry > Date.now() ||
+                  boost?.claimed ||
```

Comment:
can this 

```
                boost &&
                (boost?.expiry > Date.now() ||
                  boost?.claimed
```

You made a function called `getButtonText()` where you check all of this. Why not making something a function where you can check the state of the boost, using it here and for the `getButtonText()` also.

Wdyt ?

Reply:
done. Added a check for getting boostexpirystatus